src/components/game/GameEndingPage.test.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import GameEndingPage from './GameEndingPage';
+import Start from '../../pages/Start';
+import GameMenu from '../../pages/GameMenu';
+
+describe('GameEndingPage Component', () => {
+  it('Check that ending content is rendered', () => {
+    const { container } = render(
+      <BrowserRouter>
+        <GameEndingPage />
+      </BrowserRouter>
+    );
+
+    const endingContent = container.querySelector('.ending-content');
+    expect(endingContent).toBeInTheDocument;
+  });
+
+  it('Check that all buttons are rendered', async () => {
+    const { container } = render(
+      <BrowserRouter>
+        <GameEndingPage />
+      </BrowserRouter>
+    );
+
+    const homeButton = container.querySelector('.home-button');
+    expect(homeButton).toBeInTheDocument;
+
+    const readyButton = container.querySelector('.ready-button');
+    expect(readyButton).toBeInTheDocument;
+
+    const restartButton = container.querySelector('.restart-button');
+    expect(restartButton).toBeInTheDocument;
+  });
+
+  it('Check that the ending image is rendered', async () => {
+    const { container } = render(
+      <BrowserRouter>
+        <GameEndingPage />
+      </BrowserRouter>
+    );
+
+    const endingImage = container.querySelector('.ending-img');
+    expect(endingImage).toBeInTheDocument;
+  });
+
+  it('Check that the header is rendered', () => {
+    render(
+      <BrowserRouter>
+        <GameEndingPage />
+      </BrowserRouter>
+    );
+
+    const endingHeader = screen.getByText('Onnittelut!');
+    expect(endingHeader).toBeInTheDocument;
+  });
+
+  it('Check that all other text is rendered', () => {
+    render(
+      <BrowserRouter>
+        <GameEndingPage />
+      </BrowserRouter>
+    );
+
+    const endingText = screen.getByText(
+      'Suoritit koko polun ja saavuit määränpäähäsi!'
+    );
+    expect(endingText).toBeInTheDocument;
+  });
+
+  it('checks that the home button brings you to the start page', () => {
+    // Rendering the required pages
+    const { container } = render(
+      <BrowserRouter>
+        <Start />
+        <GameEndingPage />
+      </BrowserRouter>
+    );
+
+    const homeButton = container.querySelector('.home-button');
+    fireEvent.click(homeButton);
+    expect(container.querySelector('.start-button')).toBeInTheDocument();
+  });
+
+  it('checks that the VALMIS button bring syou to the game menu', () => {
+    // Rendering the required pages
+    const { container } = render(
+      <BrowserRouter>
+        <GameMenu />
+        <GameEndingPage />
+      </BrowserRouter>
+    );
+
+    const readyButton = container.querySelector('.ready-button');
+    expect(readyButton).toBeInTheDocument();
+
+    fireEvent.click(readyButton);
+    expect(screen.getByText('Valitse polku')).toBeInTheDocument();
+  });
```

Comment:
Näissä testeissä navigaatiota testataan väärin. Alkuperäisessä container-elementissä on jo testin alussa GameEndingPage:n elementit SEKÄ GameMenun / Start komponentin elementit. Sen voi todentaa laittamalla testin alkuun ennen fireEventtiä expect(...).not.toBeInTheDocument(); jossa elementti on siis se mikä pitäisi vasta navigoinnin jälkeen näkyä.